### ISSUE TITLE
Orchestrate doesn't handle minion error properly

### ISSUE THREAD

**Issue by basepi** [saltstack/salt#29586]
If a minion returns `"No matching sls found for <statefile> in env 'base'"` state.orchestrate stacktraces:

```
vagrant-base-trusty-amd64_master:
----------
          ID: test1
    Function: salt.state
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/local/lib/python2.7/dist-packages/salt/state.py", line 1594, in call
                  **cdata['kwargs'])
                File "/usr/local/lib/python2.7/dist-packages/salt/loader.py", line 1491, in wrapper
                  return f(*args, **kwargs)
                File "/usr/local/lib/python2.7/dist-packages/salt/states/saltmod.py", line 261, in state
                  for state_item in six.itervalues(m_ret):
                File "/usr/local/lib/python2.7/dist-packages/salt/ext/six.py", line 580, in itervalues
                  return iter(d.itervalues(**kw))
              AttributeError: 'list' object has no attribute 'itervalues'
     Started: 23:15:19.735982
    Duration: 255.657 ms
     Changes:
```

Need to handle the stacktrace and return the results somehow.

**Comment by basepi** [comment]
v2015.8.3

**Comment by basepi** [comment]
Fixed in #29603
